src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/PostgresAccessPolicyDAO.scala

```diff
@@ -437,11 +443,11 @@ class PostgresAccessPolicyDAO(protected val dbRef: DbReference,
     val pa = PolicyActionTable.syntax("pa")
     val p = PolicyTable.syntax("p")
 
-    val policyPK = PolicyPK(
-      samsql"""delete from ${PolicyActionTable as pa}
-              using ${PolicyTable as p}
-              where ${p.name} = ${id.accessPolicyName}
-              and ${p.resourceId} = (${ResourceTable.loadResourcePK(id.resource)}) returning ${p.id}""".updateAndReturnGeneratedKey().apply())
+    val policyPK = samsql"select ${p.id} from ${PolicyTable as p} where ${p.name} = ${id.accessPolicyName} and ${p.resourceId} = (${ResourceTable.loadResourcePK(id.resource)})".map(rs => PolicyPK(rs.long(1))).single().apply().getOrElse {
+      throw new WorkbenchException(s"policy record not found for $id")
+    }
+
+    samsql"delete from ${PolicyActionTable as pa} where ${pa.resourcePolicyId} = $policyPK".update().apply()
```

Comment:
sam comment as in `overwritePolicyMembersInternal`. Also I think the old code deletes all records from `PolicyActionTable` 😱 (I don't see a constraint on `pa.resourcePolicyId`)

Reply:
Looks good, could you add the missing constraint on the delete query in `overwritePolicyRolesInternal`? 
```
samsql"""delete from ${PolicyRoleTable as pr}
using ${PolicyTable as p}
where ${p.name} = ${id.accessPolicyName}
and ${p.resourceId} = (${ResourceTable.loadResourcePK(id.resource)})
and ${pr.resourcePolicyId} = ${p.id}
returning ${p.id}""".updateAndReturnGeneratedKey().apply())```